flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/match/PatternSelectFunctionRunner.scala

```diff
@@ -35,20 +36,26 @@ import org.apache.flink.util.Collector
 class PatternSelectFunctionRunner(
     name: String,
     code: String)
-  extends PatternFlatSelectFunction[Row, CRow]
-  with Compiler[PatternSelectFunction[Row, Row]]
+  extends RichPatternFlatSelectFunction[Row, CRow]
```

Comment:
Is there any reason we use a `RichPatternFlatSelectFunction` which actually wraps a `RichPatternSelectFunction`?

Reply:
We need access to the collector to be able to erase timestamp.

Reply:
OK. Make sense.

Reply:
@dawidwys I have rethought about this issue and what about always generating a `PatternFlatSelectFunction` no matter `ONE ROW PER MATCH` or `ALL ROWS PER MATCH` is specified?